website/blog/2023-05-01-package-exports-support.md

```diff
@@ -0,0 +1,229 @@
+    ...
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Here's some app code consuming the above package by importing different subpaths of `@storybook/addon-actions`.
+
+```js
+import {action} from '@storybook/addon-actions';
+// -> '@storybook/addon-actions/dist/index.js'
+
+import {action} from '@storybook/addon-actions/preview';
+// -> '@storybook/addon-actions/dist/preview.js'
+
+import helpers from '@storybook/addon-actions/src/preset/addArgsHelpers';
+// Inaccessible - not listed in "exports"!
+```
+
+The headlining features of Package Exports are:
+
+- **Package encapsulation**: Only subpaths defined in `"exports"` can be imported from outside the package — giving packages control over their public API.
+- **Subpath aliases**: Packages can define custom subpaths which map to a different file location (including via [subpath patterns](https://nodejs.org/docs/latest-v19.x/api/packages.html#subpath-patterns)) — allowing relocation of files while preserving the public API.
+- **Conditional exports**: A subpath may resolve to a different underlying file depending on environment. For example, to target `"node"`, `"browser"`, or `"react-native"` runtimes — replacing the [`"browser"` field spec](https://github.com/defunctzombie/package-browser-field-spec).
+
+:::note
+The full capabilities for `"exports"` are detailed in the [Node.js Package Entry Points spec](https://nodejs.org/docs/latest-v19.x/api/packages.html#package-entry-points).
+
+Since these features overlap with existing React Native concepts (such as [platform-specific extensions](/docs/platform-specific-code)), and since `"exports"` had been live in the npm ecosystem for some time, we reached out to the React Native community to make sure our implementation would meet developers' needs ([PR](https://github.com/react-native-community/discussions-and-proposals/pull/534), [final RFC](https://github.com/react-native-community/discussions-and-proposals/blob/main/proposals/0534-metro-package-exports-support.md)).
+:::
+
+## For app developers
+
+Package Exports can be enabled today, experimentally.
+
+- Imports against packages that depend on Package Exports features (such as [**Firebase**](https://www.npmjs.com/package/firebase) and [**Storybook**](https://www.npmjs.com/search?q=%40storybook)) should now work as designed.
+- React Native for Web projects using Metro will now be able to use the `"browser"` conditional export, removing the need for workarounds.
+
+Enabling Package Exports brings a few [edge-case breaking changes](#breaking-changes) that may affect specific projects, and which you can [test today](#validating-changes-in-your-project).
+
+**In a future React Native release, Package Exports will be enabled by default**. In a chicken-and-egg situation, React Native apps were previously a holdout for some packages to migrate to `"exports"` — or used our `"react-native"` root field escape hatch. Supporting these features in Metro will allow the ecosystem to move forward.
```

Comment:
What's the incentive for someone to enable Package Exports? Is it simply that we're going to make this default eventually and it's better to prepare earlier?

Reply:
Mostly. It immediately enables the benefits presented on lines 76-77, but admittedly for most common packages there is already a happy React Native path or packages haven't moved to `"exports"` yet. (For cases where instead workarounds in apps were previously required, see https://github.com/facebook/metro/issues/670).

When some React Native libraries start migrating to `"exports"` later this year, they should be able to point back to this post to explain why RN 0.72/0.73 or above is required to consume a breaking package release that has switched to `"exports"`.